src/slurm_plugin/fleet_manager.py

```diff
@@ -312,13 +319,26 @@ def _uses_single_az(self):
         subnet_ids = self._compute_resource_config.get("Networking", {}).get("SubnetIds", [])
         return len(subnet_ids) == 1
 
+    def _uses_subnet_prioritization(self):
+        return (
+            self._compute_resource_config.get("AllocationStrategy") == "prioritized"
```

Comment:
Can we add a valid condition where we check `prioritized` is used with `On-demand` Capacity and `capacity-optimized-prioritized` is checked with Spot Capacity type. Please update the unit test accordingly as they are passing without this change